tests/controls/textctrltest.cpp

```diff
@@ -275,51 +274,134 @@ void TextCtrlTestCase::ReadOnly()
 void TextCtrlTestCase::MaxLength()
 {
 #if wxUSE_UIACTIONSIMULATOR
+    wxUIActionSimulator sim;
+
+    if ( ms_style == wxTE_MULTILINE )
+    {
+#if defined(__WXMSW__) /*|| defined(__WXGTK3__)*/ || defined(__WXQT__)
+        delete m_text;
+        CreateText(wxTE_DONTWRAP | wxTE_RICH);
```

Comment:
@vadz I don't know what to do here, because `wxMSW` would fail without `wxTE_RICH` or `wxTE_RICH2`?
i.e. SetValue()/ChangeValue() can not write more than `maxlen` if `> 0`. and this is not what would happen if the control is created with `wxTE_RICH[2]` flags?

Reply:
Sorry, I'm not sure I understand the problem, but if there is a discrepancy between the control behaviour with and without `wxTE_RICH[2]` flag, then it should be ideally fixed or, if this is impossible, at least documented.

Reply:
First, let me correct myself: `SetValue()/ChangeValue()` always succeed in setting/changing the contents of `wxTextCtrl` (with/without `wxTE_RICH[2]`) regardless of whether `SetMaxLength()` has been called or not.

The problem is with `AppendText()` which works inconsistently (with/without `wxTE_RICH[2]`) when `SetMaxLength()` is already called with non-zero length:

1- With `wxTE_RICH[2]` flag set: `AppendText()` always succeed.
2- Without `wxTE_RICH[2]` flag: `AppendText()` fails if the maximum length is reached.

Reply:
You should be able to handle `EN_MAXTEXT`, i.e. tweak `wxTextCtrl::AdjustSpaceLimit()`, to allow appending text into plain EDIT control.

Reply:
I don't really understand what's going on here because AFAICS we already had this test for wxMSW and it seems to have worked without `wxTE_RICH` before, but I'm fine with merging this wxQt PR without dealing with any wxMSW problems anyhow.